src/DateTimeFormatter.php

```diff
@@ -4,99 +4,79 @@
+    private readonly string $format;
 
     /**
-     * Sets filter options
-     *
-     * @param array|Traversable $options
+     * A valid timezone string
      */
-    public function __construct($options = null)
-    {
-        if ($options !== null) {
-            $this->setOptions($options);
-        }
-    }
+    private readonly DateTimeZone $timezone;
 
     /**
-     * Set the format string accepted by date() to use when formatting a string
-     *
-     * @param  string $format
-     * @return self
+     * @param Options $options
      */
-    public function setFormat($format)
+    public function __construct(array $options = [])
     {
-        $this->format = $format;
-
-        return $this;
-    }
+        $this->format = $options['format'] ?? DateTimeInterface::ATOM;
 
-    /**
-     * Filter a datetime string by normalizing it to the filters specified format
-     *
-     * @param  DateTime|string|int|mixed $value
-     * @throws Exception\InvalidArgumentException
-     */
-    public function filter(mixed $value): mixed
-    {
         try {
-            $result = $this->normalizeDateTime($value);
+            $this->timezone = new DateTimeZone(
+                $options['timezone'] ?? date_default_timezone_get()
+            );
         } catch (Throwable $e) {
-            // DateTime threw an exception, an invalid date string was provided
-            throw new Exception\InvalidArgumentException('Invalid date string provided', $e->getCode(), $e);
+            throw new InvalidArgumentException('Invalid timezone provided');
         }
-
-        if ($result === false) {
-            return $value;
-        }
-
-        return $result;
     }
 
     /**
-     * Normalize the provided value to a formatted string
+     * Filter a datetime string by normalizing it to the filters specified format
      *
-     * @return string|mixed
+     * @inheritDoc
      */
-    protected function normalizeDateTime(mixed $value)
+    public function filter(mixed $value): mixed
     {
-        if ($value === '' || $value === null) {
-            return $value;
-        }
-
-        if (! is_string($value) && ! is_int($value) && ! $value instanceof DateTimeInterface) {
+        if (
+            ! (is_string($value) && $value !== '') && ! is_int($value)
+            && ! $value instanceof DateTimeInterface
+        ) {
             return $value;
         }
 
-        if (is_int($value)) {
-            //timestamp
-            $value = new DateTime('@' . $value);
-        } elseif (! $value instanceof DateTimeInterface) {
-            $value = new DateTime($value);
+        try {
+            if (! $value instanceof DateTimeInterface) {
+                $value = is_int($value) ? '@' . (string) $value : $value;
+                $value = new DateTimeImmutable($value, $this->timezone);
+            }
+        } catch (Throwable $e) {
+            throw new InvalidArgumentException('Invalid date string provided');
```

Comment:
```suggestion
        if (! $value instanceof DateTimeInterface) {
            try {
                $value = is_int($value) ? '@' . (string) $value : $value;
                $value = new DateTimeImmutable($value, $this->timezone);
            } catch (Throwable $e) {
                throw new InvalidArgumentException('Invalid date string provided');
            }
```

Nitpick - but wrapping the `try/catch` in the `if` reads more like, "If the condition is met, then try something"

Reply:
i refactored a little bit this logic because of the psalm